### ISSUE TITLE
Can no longer init sbws due to config error

### ISSUE THREAD

**Issue by pastly** [pastly/simple-bw-scanner#186]
```
(venv-editable) cranium:simple-bw-scanner mtraudt$ sbws -d /tmp/sbws init
Traceback (most recent call last):
  File "/Users/mtraudt/src/simple-bw-scanner/venv-editable/bin/sbws", line 11, in <module>
    load_entry_point('sbws', 'console_scripts', 'sbws')()
  File "/Users/mtraudt/src/simple-bw-scanner/sbws/sbws.py", line 31, in main
    configure_logging(conf)
  File "/Users/mtraudt/src/simple-bw-scanner/sbws/util/config.py", line 93, in configure_logging
    "('{}',)".format(conf['paths']['log_filepath'])
  File "/Users/mtraudt/.pyenv/versions/3.6.5/lib/python3.6/configparser.py", line 1234, in __getitem__
    return self._parser.get(self._name, key)
  File "/Users/mtraudt/.pyenv/versions/3.6.5/lib/python3.6/configparser.py", line 800, in get
    d)
  File "/Users/mtraudt/.pyenv/versions/3.6.5/lib/python3.6/configparser.py", line 455, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/Users/mtraudt/.pyenv/versions/3.6.5/lib/python3.6/configparser.py", line 507, in _interpolate_some
    option, section, rawval, ":".join(path)) from None
configparser.InterpolationMissingOptionError: Bad value substitution: option 'log_filepath' in section 'paths' contains an interpolation key 'sbws_home' which is not a valid option name. Raw value: '${sbws_home}/sbws.log'
```

Most likely `sbws_home` isn't set yetat the time `configure_logging` is called when initializing sbws for the first time.

Accompanying the fix for this, I'd like to see a test that calls `sbws init` so errors like these can be caught in the future.

**Comment by pastly** [comment]
@juga, what do you think of this?

As part of fixing this, I want to overhaul (and simplify for the user) logging configuration.

I want to ...

**Remove the config files for logging**. We'll go back to just reading config.default.ini and config.ini. This will remove a lot of flexibility from the user; however, I don't think our users were ever going to use that flexibility.

**Add to the config a section such as the following**.

```
[logging]
level = debug
to_file = yes
to_stdout = yes
format =  [%(asctime)s] [%(name)s] [%(levelname)s] %(message)s
```

Where
- level changes what messages get logged (debug, info, etc. as we are familiar with)
- to_file tells sbws to (not) log to rotating files in a log directory in ~/.sbws.
- to_stdout tells sbws to (not) log to stdout
- format is the log format string

**Keep `--log-level`, but change its possible values from ALLCAPS to lowercase**. This is more normal IMO. `--log-level` will override the `level` option in the config, if present.

**Write code that takes the above configuration settings and configures the python logging library appropriately**. This way only we have to understand the way python logging works. The user doesn't have to figure out the difference and relationship between loggers, handlers, and formatters like they currently have to if they want to understand what they're doing when they edit their config.log.ini

**Comment by juga0** [comment]
> **Remove the config files for logging**. We'll go back to just reading config.default.ini and config.ini. This will remove a lot of flexibility from the user; however, I don't think our users were ever going to use that flexibility.

sounds good, though would you still keep config.log.default.ini?
I think we should keep it. If not, we should still have logging defaults
somewhere, maybe in globals.py

> 
> **Add to the config a section such as the following**.
> 
> ```
> [logging]
> level = debug
> to_file = yes
> to_stdout = yes
> format =  [%(asctime)s] [%(name)s] [%(levelname)s] %(message)s
> ```

i'd make ``to_stdout = no`` by default.
If we leave ``to_file``, maybe we should add an option to specify where
i'd remove format and use the formats that are configured internally for
the different levels (either config.log.default.ini or globals.py).

I'd like sbws to be able to log to syslog/journal too by default, only
INFO level. That could be configured internally.

> **Keep `--log-level`, but change its possible values from ALLCAPS to lowercase**. This is more normal IMO. `--log-level` will override the `level` option in the config, if present.

ok. Some times i miss we could just do `-v` or `-vv` (to type less), but
no need to change this.

**Comment by pastly** [comment]
Thank you for the feedback Juga. I hope to work on this very soon and will keep the things you said in mind.

**Comment by pastly** [comment]
Made a lot of progress on this in branch [logging_overhaul](https://github.com/pastly/simple-bw-scanner/tree/logging_overhaul). Need to skim through it again before I'm ready for a PR. Also it probably breaks a ton of tests.

**Comment by pastly** [comment]
Closed via the series of commits ending at 43856fdbe3d242396b3c9f6533bd20b9d1c5ec27
